pyccel/parser/semantic.py

```diff
@@ -2482,8 +2514,13 @@ def _visit_FunctionDef(self, expr, **settings):
             # we annotate the body
             body = self._visit(expr.body)
 
-            args    = [self.get_variable(a.name) if isinstance(a, Variable) else self.get_function(str(a.name)) for a in args]
+            #checking if the Function don't have Return then call the garbage collector
+            if not isinstance(expr.body.args[-1][-1], Return):
+                dealloc = [Deallocate(i) for i in self._allocs]
+                body = CodeBlock([body, CodeBlock(dealloc)])
+            self._allocs = []
```

Comment:
What about functions in functions?

Reply:
i was testing with C and didn't think about this.
do we have that implemented in C ?
i will try to handle nested function

Reply:
I'm not sure if it's implemented yet but it will be at some point and we don't want to have to look for things like this when we do implement it

Reply:
okey i will try to do it